presto-main/src/test/java/io/prestosql/sql/planner/TestLocalDynamicFilter.java

```diff
@@ -0,0 +1,285 @@
+                "SELECT count() FROM lineitem, orders WHERE lineitem.orderkey = orders.orderkey " +
+                        "AND orders.custkey < 10",
+                LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED,
+                false);
+        JoinNode joinNode = PlanNodeSearcher.searchFrom(subplan.getChildren().get(0).getFragment().getRoot())
+                                            .where(node -> node instanceof JoinNode)
+                                            .findOnlyElement();
+        LocalDynamicFilter filter = LocalDynamicFilter.create(joinNode, 1).get();
+        String filterId = Iterables.getOnlyElement(filter.getBuildChannels().keySet());
+        Symbol probeSymbol = Iterables.getOnlyElement(joinNode.getCriteria()).getLeft();
+
+        filter.getTupleDomainConsumer().accept(TupleDomain.withColumnDomains(ImmutableMap.of(
+                filterId, Domain.singleValue(BIGINT, 3L))));
+        assertEquals(filter.getResultFuture().get(), TupleDomain.withColumnDomains(ImmutableMap.of(
+                probeSymbol, Domain.singleValue(BIGINT, 3L))));
+    }
+
+    @Test
+    public void testCreateMultipleCriteria()
+            throws ExecutionException, InterruptedException
+    {
+        SubPlan subplan = subplan(
+                "SELECT count() FROM lineitem, partsupp " +
+                        "WHERE lineitem.partkey = partsupp.partkey AND lineitem.suppkey = partsupp.suppkey " +
+                        "AND partsupp.availqty < 10",
+                LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED,
+                false);
+        JoinNode joinNode = PlanNodeSearcher.searchFrom(subplan.getChildren().get(0).getFragment().getRoot())
+                                            .where(node -> node instanceof JoinNode)
+                                            .findOnlyElement();
+        LocalDynamicFilter filter = LocalDynamicFilter.create(joinNode, 1).get();
+        String[] filterIds = filter.getBuildChannels()
+                                   .entrySet()
+                                   .stream()
+                                   .sorted(Comparator.comparing(e -> e.getValue()))
+                                   .map(e -> e.getKey())
+                                   .toArray(String[]::new);
+        filter.getTupleDomainConsumer().accept(TupleDomain.withColumnDomains(ImmutableMap.of(
+                filterIds[0], Domain.singleValue(BIGINT, 4L),
+                filterIds[1], Domain.singleValue(BIGINT, 5L))));
+
+        TupleDomain<Symbol> expected = TupleDomain.withColumnDomains(ImmutableMap.of(
+                new Symbol("partkey"), Domain.singleValue(BIGINT, 4L),
+                new Symbol("suppkey"), Domain.singleValue(BIGINT, 5L)));
+        assertEquals(filter.getResultFuture().get(), expected);
+    }
+
+    @Test
+    public void testCreateMultipleJoins()
+            throws ExecutionException, InterruptedException
+    {
+        SubPlan subplan = subplan(
```

Comment:
why can't you use `Plan` instead and `BasePlanTest#plan(java.lang.String, io.prestosql.sql.planner.LogicalPlanner.Stage, boolean)`
instead?

Reply:
In the case of distributed join (which I forgot to add to the unit-tests), the probe-side scan will be in a separate fragment from the join node. This should cause `LocalDynamicFilter#create` return `Optional.empty()`.
I thought that it would be easier to fragment the plan using a helper method, so that `LocalDynamicFilter#create` will operate on a plan fragment (as it does when invoked by `LocalExecutionPlanner`).

Reply:
For example, see `testCreateDistributedJoin` above:
https://github.com/prestosql/presto/pull/1702/files#diff-83eb1b35819da03bdd4c794e0047c83aR211-R227